zone_cache_reserve.go

```diff
@@ -0,0 +1,79 @@
+package cloudflare
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"time"
+)
+
+var validOnOffSettingValues = []string{"on", "off"}
+
+// ZoneCacheReserve is the structure of the API object for the
+// cache reserve setting.
+type ZoneCacheReserve struct {
+	ID         string    `json:"id,omitempty"`
+	ModifiedOn time.Time `json:"modified_on,omitempty"`
+	Value      string    `json:"value"`
+}
+
+// CacheReserveDetailsResponse is the API response for the
+// cache reserve setting.
+type CacheReserveDetailsResponse struct {
+	Result ZoneCacheReserve `json:"result"`
+	Response
+}
+
+type updateZoneCacheReserveRequest struct {
+	Value string `json:"value"`
+}
+
+type zoneCacheReserveSingleResponse struct {
+	Response
+	Result ZoneCacheReserve `json:"result"`
+}
+
+// GetZoneCacheReserve returns information about the current cache reserve settings
+//
+// API reference: https://developers.cloudflare.com/api/operations/zone-cache-settings-get-cache-reserve-setting
+func (api *API) GetZoneCacheReserve(ctx context.Context, zoneID string) (ZoneCacheReserve, error) {
+	uri := fmt.Sprintf("/zones/%s/cache/cache_reserve", zoneID)
+
+	res, err := api.makeRequestContext(ctx, http.MethodGet, uri, nil)
+	if err != nil {
+		return ZoneCacheReserve{}, err
+	}
+
+	var cacheReserveDetailsResponse CacheReserveDetailsResponse
+	err = json.Unmarshal(res, &cacheReserveDetailsResponse)
+	if err != nil {
+		return ZoneCacheReserve{}, fmt.Errorf("%s: %w", errUnmarshalError, err)
+	}
+	return cacheReserveDetailsResponse.Result, nil
+}
+
+// UpdateZoneCacheReserve updates the cache reserve setting for a zone
+//
+// API reference: https://developers.cloudflare.com/api/operations/zone-cache-settings-change-cache-reserve-setting
+func (api *API) UpdateZoneCacheReserve(ctx context.Context, zoneID string, value string) (ZoneCacheReserve, error) {
+	if !contains(validOnOffSettingValues, value) {
+		return ZoneCacheReserve{}, fmt.Errorf("invalid setting value '%s'. must be 'on' or 'off'", value)
```

Comment:
@jfchevrette, this might be validated server-side.

I am unsure what the SDK/client policy is here (@jacobbednarz might offer some guidance), but I would leave it to the backend to return an error that the client then bubbles up.

This validation would make much more sense in the Terraform provider, where avoiding this round trip would make sense.

Reply:
I have no insight into the backend so I can't comment on that. I got the validation inspiration from https://github.com/cloudflare/cloudflare-go/blob/9388783618dc500a0a8a942c407104c81de77d9a/argo.go#L53

Additional validation can of course be done in the terraform provider but this is beyond this PR

Reply:
yep, this validation can go as it's performed server side. we don't want validation in three places (lib, terraform provider and service) that all have different sources.

Reply:
I have removed the validation. Thank you